doc/source/whatsnew/v1.1.0.rst

```diff
@@ -1050,6 +1050,7 @@ Other
 - Bug in :meth:`DataFrame.plot.scatter` caused an error when plotting variable marker sizes (:issue:`32904`)
 - :class:`IntegerArray` now implements the ``sum`` operation (:issue:`33172`)
 - Bug in :class:`Tick` comparisons raising ``TypeError`` when comparing against timedelta-like objects (:issue:`34088`)
+- Fixed building from source with Python 3.8 fetching the wrong version of NumPy (:issue:`34666`)
```

Comment:
Can you move this to the 1.0.5 file?

Reply:
Are we definitely doing a 1.0.5 release?

Reply:
As long as there is a plan, it is easier to directly put it there (and when there is not a 1.0.5 done when 1.1.0 is being released, it is easy to move the content from v1.0.5.txt to the v1.1 file, but the other way around it is IMO more annoying to have to move whatsnew entries separately for backports)

Now, given the regressions, I would also say that we should do a 1.0.5